okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/Platform.java

```diff
@@ -60,6 +62,16 @@
 public class Platform {
   public static final Logger logger = Logger.getLogger(Platform.class.getName());
 
+  /**
+   * List of security providers to use in order of preference.
+   */
+  private static final String[] ANDROID_SECURITY_PROVIDERS = new String[]{
+      // See https://developer.android.com/training/articles/security-gms-provider.html
+      "com.google.android.gms.org.conscrypt.OpenSSLProvider",
+      "com.android.org.conscrypt.OpenSSLProvider",
+      "org.conscrypt.OpenSSLProvider",
```

Comment:
I don't think we are inferring anything, given that the list is clearly defined to be in preference order. So we loop through the list until we find one that we know works. That seems the canonical use of the API.

But it doesn't seem like we're going to come to agreement, and I am happy we're at least using the provider list now.